### ISSUE TITLE
Treeview event state changes slow

### ISSUE THREAD

**Issue by ShaneMcC** [DMDirc/DMDirc#745]
The treeview is very slow to receive and handle events.

After connecting to my bouncer then scrolling through all the channels to clear the notifications, I can see each channel get ui.messageNotificationColour colour set and then it loses all the colours (delayed colour-removal from scrolling over the channels)

This makes the client feel laggy on a new connect, even though everything else seems fine.

**Comment by csmith** [comment]
Yeah, I think it's just the fact that we're throwing around millions of events when processing backbuffers for lots of channels.

The unread message events should either be batched up or throttled, especially when we know we're receiving backbuffer content from a bouncer
